core/domain/suggestion_services.py

```diff
@@ -599,6 +606,211 @@ def get_translation_suggestions_waiting_longest_for_review_per_lang(
+        # Convert the MatchObject to a string.
+        rte_tag_string = rte_tag.group(0)
+        # Get the name of the noninteractive tag (ex. math).
+        replace_string = rte_tag_string.split('-')[2].split(' ')[0]
+        if replace_string[-1] == '>':
+            replace_string = replace_string[:-1]
+        replace_string = replace_string.upper()
+        replace_string = '%s%s%s' % (' [', replace_string, '] ')
+        return replace_string
+    # Replace the rte tags by their name in square brackets.
+    html_content_string = re.sub(
+        '<([^>]+)>', _replace_rte_tag, html_content_string)
+    # Remove any leading or trailing whitespace.
+    resulting_plain_text = html_content_string.strip()
+    return resulting_plain_text
+
+
+def create_reviewable_suggestion_email_info_from_suggestion(suggestion):
+    """Creates an object with the key information needed to notify reviewers or
+    admins that the given suggestion needs review.
+    Note: the suggestions must be suggestions that are available for review on
+    the Contributor Dashboard.
+
+    Args:
+        suggestion: Suggestion. The suggestion used to create the
+            ReviewableSuggestionEmailInfo object.
+
+    Returns:
+        ReviewableSuggestionEmailInfo. The corresponding reviewable suggestion
+        email info.
+
+    Raises:
+        Exception. The suggestion must be in review.
+        Exception. The suggestion type must be offered on the Contributor
+            Dashboard.
+    """
+    if suggestion.status != suggestion_models.STATUS_IN_REVIEW:
+        raise Exception(
+            'Expected suggestion status to be in review for '
+            'ReviewableSuggestionEmailInfo object creation, recieved '
+            'suggestion status: %s' % suggestion.status)
+    if suggestion.suggestion_type != (
+            suggestion_models.SUGGESTION_TYPE_TRANSLATE_CONTENT) and (
+                suggestion.suggestion_type != (
+                    suggestion_models.SUGGESTION_TYPE_ADD_QUESTION)):
+        raise Exception(
+            'Expected suggestion type to be offered on the Contributor '
+            'Dashboard for ReviewableSuggestionEmailInfo object creation, '
+            'recieved suggestion type: %s' % suggestion.suggestion_type)
+    html_content_strings = suggestion.get_all_html_content_strings()
+    html_content_string = ''
+    if suggestion.suggestion_type == (
+            suggestion_models.SUGGESTION_TYPE_TRANSLATE_CONTENT):
+        # The translation html is always the first element in the list of
+        # translation suggestion html content strings.
+        html_content_string = html_content_strings[0]
+    elif suggestion.suggestion_type == (
+            suggestion_models.SUGGESTION_TYPE_ADD_QUESTION):
+        # The question content html is always the last element in the list of
+        # question suggestion html content strings.
+        html_content_string = html_content_strings[-1]
```

Comment:
else raise.

Or better: have a dict constant that maps each suggestion type to a lambda function, and just read off the value from that.

Reply:
Looks resolved.